src/compiler/parsing/watf/index.js

```diff
@@ -2,14 +2,15 @@
 
 const {parse} = require('./grammar');
 const {tokenize} = require('./tokenizer');
+const wastIdentifierToIndex = require('../../transform/wast-identifier-to-index');
 
-function parseSource(source: string): Program {
+export function parseSource(source: string): Program {
   const tokens = tokenize(source);
 
   // We pass the source here to show code frames
   const ast = parse(tokens, source);
 
+  wastIdentifierToIndex.transform(ast);
```

Comment:
I decided to add the transformation right after the parsing, it will keep things more consistent. But I had to modify some unrelated tests to make them pass.